models/packages/descriptor.go

```diff
@@ -110,9 +110,12 @@ func GetPackageDescriptor(ctx context.Context, pv *PackageVersion) (*PackageDesc
 	if err != nil {
 		return nil, err
 	}
-	repository, err := repo_model.GetRepositoryByID(ctx, p.RepoID)
-	if err != nil && !repo_model.IsErrRepoNotExist(err) {
-		return nil, err
+	var repository *repo_model.Repository
+	if p.RepoID > 0 {
+		repository, err = repo_model.GetRepositoryByID(ctx, p.RepoID)
```

Comment:
What if the "linked" repository is deleted? Then the package becomes 404?

Reply:
Looks like this is a bug not only for composor but all packages which linked to a repository.

Reply:
#33634

Reply:
#33653 should fix all similiar bugs. So I think we can mark this as resolved here.

Reply:
I think it should tolerate non-existing repo here and "Add API to support link package to repository and unlink it #33481"


Does it make sense to report 500 error when a repo doesn't exist?


![image](https://github.com/user-attachments/assets/9d732147-37eb-4623-8fe9-2529aa46c07a)

Reply:
Looks like line 116, I think the error repo doesn't exist has been ignored.

Reply:
8115cf1195 will allow ignoring non-existing repo